docs/cpp_engine_architecture.md

```
@@ -0,0 +1,174 @@
+
+---
+
+## 📡 3. Command Protocol
+
+### 🔹 MOVES Command
+
+**Purpose:** Get valid moves for a selected piece
+
+**Input:**
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)